src/GraphQL/Execution/ParallelExecutionStrategy.cs

```diff
@@ -21,10 +21,7 @@ public class ParallelExecutionStrategy : ExecutionStrategy
         /// Nodes that return a <see cref="IDataLoaderResult"/> will execute once all other pending nodes
         /// have been completed.
         /// </summary>
-        protected override Task ExecuteNodeTreeAsync(ExecutionContext context, ObjectExecutionNode rootNode) => ExecuteNodeTreeAsync(context, rootNode);
```

Comment:
Interesting. Ho did it work before? `protected ExecuteNodeTreeAsync` calls `public ExecuteNodeTreeAsync`. I mean compiler resolves overload with less concrete signature (ExecutionNode vs ObjectExecutionNode). 🤔

Reply:
Well before the `ExecuteNodeTreeAsync(ExecutionContext, ObjectExecutionNode)` existed because it was abstract on `ExecutionStrategy` and hence required -- `ExecutionStrategy.ExecuteAsync` would call this method.

But within `ParallelExecutionStrategy` it just immediately passed control to `ExecuteNodeTreeAsync(ExecutionContext, ExecutionNode)`, which only existed because the subscription strategy (which was a derived class) needed access to the method without being restricted to an `ObjectExecutionNode`.  So I just changed the signature of `ExecutionStrategy.ExecuteNodeTreeAsync` which cleaned up everything, without really any negative effect.  Doing so allowed `SerialExecutionStrategy.ExecuteNodeTreeAsync` to serve subscription needs when desired.

With that fixed, now I could derive from `SerialExecutionStrategy` or `ParallelExecutionStrategy` to create `SubscriptionExecutionStrategy`, I cannot derive from both, so I added the `ExecuteNodeTreeAsync` to the `IExecutionStrategy` interface so that I could pass any `IExecutionStrategy` to a `SubscriptionExecutionStrategy` instance.

So `new SubscriptionExecutionStrategy(SerialExecutionStrategy.Instance)` will create a subscription with a serial underlying execution strategy.

Reply:
> I mean compiler resolves overload with less concrete signature (ExecutionNode vs ObjectExecutionNode).

Uhh, no idea.  Sure looks like it calls itself, doesn't it?  Pretty sure it was in use, called by `ExecutionStrategy.ExecuteAsync` however.

Reply:
>  Sure looks like it calls itself, doesn't it?

I'm talking exactly about that.

Reply:
Compiler prefers non-virtual instance method over overridden one.